app/models/orcid_work.rb

```diff
@@ -0,0 +1,61 @@
+class OrcidWork < OrcidResource
+  def to_json
+    external_ids = %i[isbn issn doi url]
+    url = publication.open_access_url || publication.url
+
+    work = {
+      title: {
+        title: publication.title,
+        subtitle: publication.secondary_title,
+      },
+      "journal-title": publication.journal_title,
+      "short-description": publication.abstract,
+      type: 'journal-article',
+      "publication-date": {
+        year: publication.published_on.year,
+        month: publication.published_on.month,
+        day: publication.published_on.day
+      },
+      url: url
+    }
+
+    external_ids.each do |type|
+      if publication.send(type).present?
+        relationship = ([:doi, :url].include?(type) ? 'self' : 'part-of')
+        orcid_type = (type == :url ? :uri : type)
+        work[:"external-ids"] = { "external-id": [] } unless work[:"external-ids"].present?
+        work[:"external-ids"][:"external-id"] << {
+              "external-id-type": orcid_type.to_s,
+              "external-id-value": (type == :url ? url : publication.send(type).to_s),
+              "external-id-relationship": relationship
+        }
+      end
+    end
+
+    publication.authorships.each do |ext_author|
```

Comment:
I'm not certain about this, but here we may want to use the metadata database concept of `Contributor` instead of `Authorship`. That will give us a list of the names of all of the authors on the work rather than just the Penn State-affiliated authors.